design/backup-repo-config.md

```diff
@@ -0,0 +1,163 @@
+
+Below are the struct for the configMap:  
+``` golang
+type RepoConfig struct {
+	CacheLimitMB int `json:"cacheLimitMB,omitempty"`
+	EnableCompression int `json:"enableCompression,omitempty"`
+}
+
+type RepoConfigs struct {
+	Configs map[string]RepoConfig `json:"configs"`
+}
+```
+
+### Configurations
+
+With the above mechanisms, any kind of configuration could be added. Here list the configurations defined at present:  
+```cacheLimitMB```: specifies the size limit(in MB) for the local data cache. The more data is cached locally, the less data may be downloaded from the backup storage, so the better performance may be achieved. Practically, users can specify any size that is smaller than the free space so that the disk space won't run out. This parameter is for each repository connection, that is, users could change it before connecting to the repository. If a backup repository doesn't use local cache, this parameter will be ignored. For Kopia repository, this parameter is supported.  
+```enableCompression```: specifies to enable/disable compression for a backup repsotiory. Most of the backup repositories support the data compression feature, if it is not supported by a backup repository, this parameter is ignored. Most of the backup repositories support to dynamically enable/disable compression, so this parameter is defined to be used whenever creating a write connection to the backup repository, if the dynamically changing is not supported, this parameter will be hornored only when initializing the backup repository. For Kopia repository, this parameter is supported and can be dynamically modified.  
+
+### Sample
+Below is an example of the BackupRepository configMap with the configurations:  
+json format:  
+```json
+{
+  "configs": {
+    "repo-type-1": {
+      "cacheLimitMB": 2048,
+      "enableCompression": true    
+    },
+    "repo-type-2": {
+      "cacheLimitMB": 1024,
+      "enableCompression": false    
+    }
+  }
+}
+```  
+yaml format:  
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: <config-name>
+  namespace: velero
+data:
+  configs: |
+    {
+      "repo-type-1": {
+        "cacheLimitMB": 2048,
+        "enableCompression": true    
+      },
+      "repo-type-2": {
+        "cacheLimitMB": 1024,
+        "enableCompression": false    
+      }
+    }    
+```
+
+To create the configMap, users need to save something like the above sample to a file and then run below commands:  
+```
+kubectl create cm <config-name> -n velero --from-file=<json file name>
+kubectl label cm <config-name> -n velero backup-repository-config=<repository type>
+```  
+Or
+```
+kubectl apply -f <yaml file name>
+```  
+
+## Installation
+In order to set the ```--backup-repository-config``` parameter to the velero server, a new installation parameter ```--backup-repository-config``` will be added. An example of the installation as below:
+```
+velero install --backup-repository-config <configmap-name>
```

Comment:
This could be optional b/c we can hard code a configmap name and pass it to the velero server, and the user will only need to create the configmap with the hard-coded name.

If we took this approach ^ we may not need to update the installation flow, bearing in mind that when velero is installed it's very possible the namespace of velero is not created.

Reply:
We will remove this installation flag and keep the velero server flag.
By default installation using velero CLI, the server flag is not set; users need to modify the deployment to set the flag and wait velero server to restart.